MeesGame/Gameplay/UIObjects/ScrollBar.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Microsoft.Xna.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Xna.Framework.Graphics;
+
+namespace MeesGame
+{
+    public class ScrollBar : UIObject
+    {
+        private readonly Color barColor = Color.Aqua;
+        private readonly Color backColor = Color.Beige;
+        //stores the total length of the scrollbar.
+        private int totalElementsSize;
+        //initialize scrollbar on top
+        private int scolldistanceStartDrag;
+        private int ScrollDistance {
+            get
+            {
+                if (totalElementsSize - Rectangle.Height != 0)
+                    return ((UIList)parent).ElementsOffset * (Rectangle.Height - (int)(Rectangle.Height / (double)totalElementsSize * Rectangle.Height)) / (totalElementsSize - Rectangle.Height);
+                return 0;
+            }
+            set
+            {
+                if (totalElementsSize - Rectangle.Height != 0)
+                    ((UIList)parent).ElementsOffset = (int)(value * (totalElementsSize - Rectangle.Height) / (double)(Rectangle.Height - BarRectangle.Height));
+                else
+                    ((UIList)parent).ElementsOffset = 0;
+            }
+        }
+        //a solid color is inserted into this texture for painting a solid color
+        private Texture2D emptyTexture;
+
+        public Rectangle BarRectangle
+        {
+            get
+            {
+                if (totalElementsSize <= Rectangle.Height)
+                    return Rectangle;
+                else
+                {
+                    return new Rectangle(Rectangle.X, Rectangle.Y + ScrollDistance, Rectangle.Width, (int)(Rectangle.Height / (double)totalElementsSize * Rectangle.Height));
```

Comment:
`(int)(Rectangle.Height / (double)totalElementsSize * Rectangle.Height)`. Ik zou dit in een aparte property zetten want je gebruikt dit meerdere keren in je code